src/app/core/layout/header/header.component.ts

```diff
@@ -0,0 +1,49 @@
+import { Component, OnInit } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { RouterLink } from '@angular/router';
+import { AuthenticationService } from '../../services/authentication.service';
+import { MmsaiAccountButtonComponent } from '../../components/mmsai-account-button/mmsai-account-button.component';
+import { MenuItem } from 'primeng/api';
+import { Store } from '@ngrx/store';
+import { signOut } from '../../states/auth/auth.actions';
+
+@Component({
+  selector: 'app-header',
+  standalone: true,
+  imports: [CommonModule, RouterLink, MmsaiAccountButtonComponent],
+  templateUrl: './header.component.html',
+  styleUrls: ['./header.component.scss'],
+})
+export class HeaderComponent implements OnInit {
+  isLoggedIn!: boolean;
+  displayName?: string | null;
+  menuItems!: MenuItem[];
+
+  constructor(
+    private authenticationService: AuthenticationService,
+    private store: Store,
+  ) {}
+
+  ngOnInit() {
+    this.authenticationService.getUser().subscribe((user) => {
+      if (user) {
+        this.isLoggedIn = true;
+        this.displayName = user.displayName ? user.displayName : user.email;
+        this.menuItems = [
+          {
+            label: 'Sign Out',
+            command: () => {
+              this.store.dispatch(signOut());
+            },
+          },
+        ];
+      } else {
+        this.isLoggedIn = false;
+        this.menuItems = [
+          { label: 'Sign In', routerLink: '/sign-in' },
+          { label: 'Sign Up', routerLink: '/sign-up' },
+        ];
```

Comment:
So you're saying I should add the items to the component as html elements and hide/show the items using the loggedIn variable in ngIf? I agree this would be better in this case, but I don't think I can do that. I'm using primeNGs menu component for the popup, which requires such an object: https://primeng.org/menu

I tried using the `visible` prop for each menu item, but then I couldn't get the items to rerender properly when the loggedIn state changed.

Reply:
Oh I did not realize that you rely on external components. I would have searched for some manual approach but I believe it's not possible. All good then.